pkg/workceptor/workceptor.go

```diff
@@ -16,17 +17,29 @@ import (
 
 	"github.com/ansible/receptor/pkg/certificates"
 	"github.com/ansible/receptor/pkg/controlsvc"
+	"github.com/ansible/receptor/pkg/logger"
 	"github.com/ansible/receptor/pkg/netceptor"
 	"github.com/ansible/receptor/pkg/randstr"
 	"github.com/ansible/receptor/pkg/utils"
 	"github.com/golang-jwt/jwt/v4"
 )
 
+// Netceptor is a interface to decouple workceptor from netceptor.
+// it includes only the functions that netceptor uses.
+// the interface name is work in progress.
+type Netceptor interface {
```

Comment:
The golang naming convention for interfaces is to use an `er` postfix, and while this works in most cases `Netceptorer` is difficult to read. 
I originally had the interface name as a name pairing, outlining the interface relationship they represent, however `WorkceptorsNetceptor` while intuitive is overly verbose. 
Other languages have the `i` convention, and while `INetceptor` would be easy to understand, it breaks the golang naming convention and is overly verbose, with `i` and `interface` both being written on the name line.

@shanemcd had the suggestion of using just `Netceptor`. It is the cleanest to look at however, not the most obvious. The reader needs to understand that in order to call the concrete netceptor then namespace syntax needs to be used, e.g. `netceptor.Netceptor`.

I am leaning toward following Shanes suggestion, and hoping that any readers of the code base will pick up that the `interface` keyword is on this line and understand its special meaning.